pyro/infer/svgd.py

```diff
@@ -0,0 +1,154 @@
+import math
+import torch
+
+import pyro
+from pyro.infer import Trace_ELBO
+
+
+def vectorize(fn, num_particles, max_plate_nesting):
+    def _fn(*args, **kwargs):
+        with pyro.plate("num_particles_vectorized", num_particles, dim=-max_plate_nesting - 1):
+            return fn(*args, **kwargs)
+    return _fn
+
+
+class SVGDKernel(object):
```

Comment:
could also call this a `SteinKernel` or the like (as in KSD = kernelized stein discrepancy)